spmimage/decomposition/dict_learning.py

```diff
@@ -42,7 +45,100 @@ def sparse_encode_with_mask(X, dictionary, mask, **kwargs):
     """
     code = np.zeros((X.shape[0], dictionary.shape[0]))
     for idx in range(X.shape[0]):
-        code[idx, :] = sparse_encode(X[idx, :][mask[idx, :] == 1].reshape(1, -1),
-                                     dictionary[:, mask[idx, :] == 1],
-                                     **kwargs)
+        code[idx, :] = sklearn.decomposition.sparse_encode(
```

Comment:
@zalgo-edu In this function, `sklearn.decomposition.sparse_encode` is used.
This is the function in scikit learn. However, we should use `sparse_encode` defined by override here.